slynk/slynk-gray.lisp

```diff
@@ -99,6 +99,17 @@
 (defmethod stream-line-column ((stream sly-output-stream))
   (with-sly-output-stream stream column))
 
+(defmethod reset-stream-line-column ((stream sly-output-stream))
+  (with-sly-output-stream stream (setf column 0)))
+
+(defmethod reset-stream-line-column ((stream t))
```

Comment:
Is this method ever legitimally called?  Don't we simply want a `NO-APPLICABLE-METHOD` to be raised if this ever happens?

Reply:
I am not sure if this method is ever legitimately called by the current implementations. It's just a fail safe mechanism. If you had an implementation (like SBCL) that uses slynk-gray but does not actually use a slynk-gray stream for *standard-output*, and you didn't have this (or its own method like SBCL gets), you'd get a broken REPL.

Reply:
Hmmm, i'm not sure.  I've never liked these catch-all methods.  

But are you saying if we remove this method for `T` and the method for `SB-SYS:FD-STREAM` we get an error?  If so, can you post the backtrace of this error?

Reply:
Never mind, the error is just probably "no applicable method for SB-SYS:FD-STREAM"

Reply:
> Hmmm, i'm not sure. I've never liked these catch-all methods.

In this case, this catch-all method just makes the new function a no-op, and restores the previously existing default behavior. It ensures that if support for an SBCL-like implementation were added, it wouldn't immediately break.

> But are you saying if we remove this method for T and the method for SB-SYS:FD-STREAM we get an error? If so, can you post the backtrace of this error?

Right. Because SBCL (despite loading slynk-gray.lisp) isn't actually using a stream of the class `sly-output-stream` for output. Here's the error:

```lisp
There is no applicable method for the generic function
  #<STANDARD-GENERIC-FUNCTION SLYNK-GRAY::RESET-STREAM-LINE-COLUMN (1)>
when called with arguments
  (#<SB-SYS:FD-STREAM for "socket 127.0.0.1:42615, peer: 127.0.0.1:57544" {1002389DE3}>).
   [Condition of type SB-PCL::NO-APPLICABLE-METHOD-ERROR]

Restarts:
 0: [RETRY] Retry calling the generic function.
 1: [*ABORT] Return to SLY's top level.
 2: [ABORT] abort thread (#<THREAD "sly-channel-1-mrepl-remote-1" RUNNING {100263B4D3}>)

Backtrace:
 0: ((:METHOD NO-APPLICABLE-METHOD (T)) #<STANDARD-GENERIC-FUNCTION SLYNK-GRAY::RESET-STREAM-LINE-COLUMN (1)> #<SB-SYS:FD-STREAM for "socket 127.0.0.1:42615, peer: 127.0.0.1:57544" {1002389DE3}>) [fast-me..
 1: (SB-PCL::CALL-NO-APPLICABLE-METHOD #<STANDARD-GENERIC-FUNCTION SLYNK-GRAY::RESET-STREAM-LINE-COLUMN (1)> (#<SB-SYS:FD-STREAM for "socket 127.0.0.1:42615, peer: 127.0.0.1:57544" {1002389DE3}>))
 2: (SLYNK::FLUSH-LISTENER-STREAMS #<SLYNK-MREPL::MREPL mrepl-1-1>)
 3: ((FLET "CLEANUP-FUN-1" :IN SLYNK-MREPL::MREPL-EVAL)) [cleanup]
 4: (SLYNK-MREPL::MREPL-EVAL #<SLYNK-MREPL::MREPL mrepl-1-1> "(print \"hey\")")
 5: (SLYNK:PROCESS-REQUESTS NIL)
 6: ((LAMBDA NIL :IN SLYNK::SPAWN-CHANNEL-THREAD))
 7: ((LAMBDA NIL :IN SLYNK::SPAWN-CHANNEL-THREAD))
 8: (SLYNK-SBCL::CALL-WITH-BREAK-HOOK #<FUNCTION SLYNK:SLYNK-DEBUGGER-HOOK> #<FUNCTION (LAMBDA NIL :IN SLYNK::SPAWN-CHANNEL-THREAD) {100263BFEB}>)
 9: ((FLET SLYNK-BACKEND:CALL-WITH-DEBUGGER-HOOK :IN "/home/dieggsy/.emacs.d/straight/repos/sly/slynk/backend/sbcl.lisp") #<FUNCTION SLYNK:SLYNK-DEBUGGER-HOOK> #<FUNCTION (LAMBDA NIL :IN SLYNK::SPAWN-CHAN..
10: ((LAMBDA NIL :IN SLYNK::CALL-WITH-LISTENER))
11: (SLYNK::CALL-WITH-BINDINGS ((*PACKAGE* . #<PACKAGE "COMMON-LISP-USER">) (*DEFAULT-PATHNAME-DEFAULTS* . #P"/home/dieggsy/") (* . #1=#<SB-SYS:FD-STREAM for "socket 127.0.0.1:42615, peer: 127.0.0.1:57544..
12: ((LAMBDA NIL :IN SLYNK::SPAWN-CHANNEL-THREAD))
13: ((FLET SB-UNIX::BODY :IN SB-THREAD::RUN))
14: ((FLET "WITHOUT-INTERRUPTS-BODY-10" :IN SB-THREAD::RUN))
15: ((FLET SB-UNIX::BODY :IN SB-THREAD::RUN))
16: ((FLET "WITHOUT-INTERRUPTS-BODY-3" :IN SB-THREAD::RUN))
17: (SB-THREAD::RUN)
18: ("foreign function: call_into_lisp")
19: ("foreign function: funcall1")
```

Reply:
> In this case, this catch-all method just makes the new function a no-op, and restores the previously existing default behavior. It ensures that if support for an implementation that follows SBCL's current behavior were added, it wouldn't immediately break.

New CL implementations aren't exactly popping up each minute.  And if there were and were added to SLY, I _would_ like it to error verysoon and then decide.  It's much worse to silently produce buggy behaviour.

Reply:
> New CL implementations aren't exactly popping up each minute. And if there were and were added to SLY, I would like it to error verysoon and then decide. It's much worse to silently produce buggy behaviour.

This seems pretty reasonable to me. I've removed the `T` method and force pushed to my branch.